include/oneapi/dpl/pstl/hetero/numeric_impl_hetero.h

```diff
@@ -223,6 +223,18 @@ struct adjacent_difference_wrapper
 {
 };
 
+template <typename _Op, typename _It1ValueT, typename _It2ValueTRef>
+struct __pattern_adjacent_difference_op_caller_fn
```

Comment:
It seems the `class __transform_functor` does the same...

Reply:
Fixed